components/button/button-add.js

```diff
@@ -0,0 +1,131 @@
+import '../colors/colors.js';
+import '../icons/icon.js';
+import '../tooltip/tooltip.js';
+import { css, html, LitElement } from 'lit';
+import { FocusMixin } from '../../mixins/focus/focus-mixin.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { labelStyles } from '../typography/styles.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+
+/**
+ * A component for quickly adding items to a specific locaiton.
+ */
+class ButtonAdd extends FocusMixin(LocalizeCoreElement(LitElement)) {
+	static get properties() {
+		return {
+			/**
+			 * The text to be shown in the tooltip as a hint when visible-text is false
+			 * @type {string}
+			 */
+			label: { type: String },
+			/**
+			 * The text to show in the button when visible-text is true
+			 * @type {string}
+			 */
+			text: { type: String },
+			/**
+			 * When true, show the button with icon and visible text. When false, only show icon.
+			 * @type {boolean}
+			 */
+			visibleText: { type: Boolean, reflect: true, attribute: 'visible-text' }
+		};
+	}
+
+	static get styles() {
+		return [labelStyles, css`
+			:host {
+				--d2l-button-add-line-style: solid;
+			}
+			button {
+				align-items: center;
+				background-color: transparent;
+				border: 0;
+				box-shadow: none;
+				display: flex;
+				font-family: inherit;
+				justify-content: center;
+				outline: none;
+				padding: 0;
+				position: relative;
+				white-space: nowrap;
+				width: 100%;
+			}
+
+			.line {
+				border-top: 1px var(--d2l-button-add-line-style) var(--d2l-color-mica);
+				margin: 3px 0; /** hover/click target */
```

Comment:
Is this needed if the line is within the button? I would have thought that the click target would effectively be a rectangle of the button, including the whitespace inside the button above and below the line.

Reply:
It is - the click target for the line does end up being just the line and so we want that buffered a bit (3 px on each side). The design also specifies that we don't want the click target for the line to be the same height as the "+" / "+ Add Item" part.